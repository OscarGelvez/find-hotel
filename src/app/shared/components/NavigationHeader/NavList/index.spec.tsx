import '@testing-library/jest-dom';

import { cleanup, getAllByTestId, render } from '@testing-library/react';
import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import { NavList } from './index';

afterEach(cleanup);

const items = [
  { label: 'Habitaciones', url: '/rooms' },
  { label: 'Reservas', url: '/bookings' },
];

it('rederizar el componente NavList sin errores', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <BrowserRouter>
      <NavList items={[]} />
    </BrowserRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('renderiza el componente NavList cuando se envían items', () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <NavList items={items} />
    </BrowserRouter>
  );

  const nodes = getAllByTestId(getByTestId('nav-list'), 'nav-item-li');
  expect(nodes.length).toBe(2);
});

it('Compara snapshot del componente NavList cuando se envían items', () => {
  const element = renderer
    .create(
      <BrowserRouter>
        <NavList items={items} />
      </BrowserRouter>
    )
    .toJSON();
  expect(element).toMatchSnapshot();
});
