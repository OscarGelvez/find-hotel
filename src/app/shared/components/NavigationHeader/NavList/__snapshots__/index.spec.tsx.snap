// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compara snapshot del componente NavList cuando se env√≠an items 1`] = `
<ul
  className="sc-AxiKw jmqPVm"
  data-testid="nav-list"
>
  <li
    data-testid="nav-item-li"
  >
    <div
      className="sc-AxirZ jxvKur"
      data-testid="nav-item"
    >
      <a
        className="sc-AxjAm jqjyPd"
        data-testid="link-nav-item"
        href="/rooms"
        onClick={[Function]}
      >
        Habitaciones
      </a>
    </div>
  </li>
  <li
    data-testid="nav-item-li"
  >
    <div
      className="sc-AxirZ jxvKur"
      data-testid="nav-item"
    >
      <a
        className="sc-AxjAm jqjyPd"
        data-testid="link-nav-item"
        href="/bookings"
        onClick={[Function]}
      >
        Reservas
      </a>
    </div>
  </li>
</ul>
`;
